python/pyspark/sql/types.py

```diff
@@ -1678,37 +1678,105 @@ def from_arrow_schema(arrow_schema):
          for field in arrow_schema])
 
 
-def _check_dataframe_localize_timestamps(pdf):
+def _old_pandas_exception_message(e):
+    """ Create an error message for importing old Pandas.
     """
-    Convert timezone aware timestamps to timezone-naive in local time
+    msg = "note: Pandas (>=0.19.2) must be installed and available on calling Python process"
+    return "%s\n%s" % (e.message, msg)
```

Comment:
I am quite sure the `message` attribute does not exist in Python 3:

```python
>>> try:
...     import foo
... except ImportError as e:
...     dir(e)
...
['__cause__', '__class__', '__context__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__suppress_context__', '__traceback__', 'args', 'msg', 'name', 'path', 'with_traceback']
```

Probably, we should use `from pyspark.util import _exception_message` (https://github.com/apache/spark/pull/16845).

Reply:
Oops, thanks!
We need to fix [dataframe.py#L1905](https://github.com/apache/spark/blob/3db2bea22c6bd240c871cb4580fbbf0ee3b659b4/python/pyspark/sql/dataframe.py#L1905) as well.